src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -63,29 +76,29 @@ public void setSpinSpeed(double newSpeed) throws IllegalStateException {
    * Detects if the intake is retracted.
    */
   public boolean isRetracted() {
-    return solenoid.get() == DoubleSolenoid.Value.kReverse;
+    return topSolenoid.get() == DoubleSolenoid.Value.kReverse;
   }
 
   /**
    * Detects if the intake is extended.
    */
   public boolean isExtended() {
-    return solenoid.get() == DoubleSolenoid.Value.kForward;
+    return topSolenoid.get() == DoubleSolenoid.Value.kForward;
   }
 
   /**
    * Detects if the intake is in neutral extension.
    */
   public boolean isNeutralExtension() {
-    return solenoid.get() == DoubleSolenoid.Value.kOff;
+    return topSolenoid.get() == DoubleSolenoid.Value.kOff;
   }
 
   /**
-   * Toggles the {@link DoubleSolenoid.Value} between Extended and Retracted. If in neutral toggles
-   * it to Retracted.
+   * Toggles the {@link DoubleSolenoid.Value}s between Extended and Retracted. If in neutral toggles
+   * them to Retracted.
    */
   public void toggle() {
-    switch (solenoid.get()) {
+    switch (topSolenoid.get()) {
```

Comment:
change this method to something like this pseudocode:

if is extended, retract
else, extend

ALSO, see comment on retract

Reply:
Wasn't the default (if it's neutral) to retract it, not extend it?

Reply:
Yes, do this instead:
if is retracted: extend
else: retract